tests/ecosystem/upgrade/test_upgrade_ocp.py

```diff
@@ -0,0 +1,156 @@
+import logging
+import json
+
+from ocs_ci.ocs import ocp
+from ocs_ci.framework import config
+from ocs_ci.utility.utils import TimeoutSampler
+from ocs_ci.framework.testlib import ManageTest
+from ocs_ci.ocs.cluster import CephCluster, CephHealthMonitor
+
+logger = logging.getLogger(__name__)
+
+
+# TODO: add image type validation (ga to ga , nightly to nightly, newer than current etc.)
+
+
+class TestUpgradeOCP(ManageTest):
+    """
+    1. check cluster health
+    2. check OCP version
+    3. perform OCP upgrade
+    4. check all OCP ClusterOperators
+    5. check OCP version
+    5. monitor cluster health
+    """
+    ocp_o = ocp.OCP()
+
+    def test_upgrade_ocp(self):
+        """
+
+        Tests OCS stability when upgrading OCP
+
+        """
+
+        ceph_cluster = CephCluster()
+        with CephHealthMonitor(ceph_cluster):
+
+            target_image = config.UPGRADE['ocp_upgrade_version']
+            image_path = config.UPGRADE['ocp_upgrade_path']
+            self.cluster_operators = self.get_all_cluster_operators()
+            logger.info(f" oc version: {self.get_current_oc_version()}")
+            self.get_all_cluster_operators()
+
+            # Upgrade OCP
+
+            self.upgrade_ocp(image=target_image, image_path=image_path)
+
+            # Wait for upgrade
+            for ocp_operator in self.cluster_operators:
+                logger.info(f"Checking upgrade status of {ocp_operator}:")
+                ver = self.get_cluster_operator_version(ocp_operator)
+                logger.info(f"current {ocp_operator} version: {ver}")
+                for sampler in TimeoutSampler(
+                    timeout=2700,
+                    sleep=60,
+                    func=self.check_upgrade_completed,
+                    target_version=target_image
+                ):
+                    logger.debug(f"sampler: {sampler}")
```

Comment:
this is confusing.
sampler will return bool if upgrade completed or not so I think it will be better to log the current version and what we are expecting in case sampler is False

Reply:
this info (currnet version) provided by function:check_upgrade_completed

Reply:
Also think that this log should be better. Something like.
```python
logger.debug(f"OCP upgrade {'completed!' if sampler else 'didn't completed yet!'}")
```